### ISSUE TITLE
ForeachSourceInspection on multi-dimensional arrays

### ISSUE THREAD

**Issue by arcticlinux** [kalessil/phpinspectionsea#933]
If you reference the return of returnMultiArray in the same class file, the inspection is not triggered:

```
class TestMultiDoc
{

    /**
     * @return array[]
     */
    public function returnMultiArray(): array
    {
        $firstArray = [];
        $secondArray = [];
        return [$firstArray, $secondArray];
    }




    public function testHinting(): void
    {
        list($firstArray, $secondArray) = $this->returnMultiArray();

        foreach($firstArray as $first)
        {
            unset($first);
        }

        foreach($secondArray as $second)
        {
            unset($second);
        }
    }
}
```

But if you reference it in another class, it throws the ForeachSourceInspection warning:
"Expressions' type was not recognized, please check type hints." on $firstArray and $secondArray

```
<?php


class TestMultiDocUse
{
    public function testHinting(): void
    {
        list($firstArray, $secondArray) = (new TestMultiDoc())->returnMultiArray();

        foreach($firstArray as $first)
        {
            unset($first);
        }

        foreach($secondArray as $second)
        {
            unset($second);
        }
    }
}
```

**Comment by kalessil** [comment]
Weird, will check. Thank you Michael!

**Comment by kalessil** [comment]
Checked, that's something to be reported to PhpStorm team - we cannot fix this on our end, unfortunately.
